resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -100,11 +98,53 @@ protected DnsNameResolverContext(DnsNameResolver parent,
         allowedQueries = maxAllowedQueries;
     }
 
-    protected Promise<T> promise() {
-        return promise;
+    void resolve(Promise<T> promise) {
+        resolve(promise, parent.searchDomains().length > 0 && !hostname.endsWith("."));
     }
 
-    void resolve() {
+    private void resolve(Promise<T> promise, boolean trySearchDomain) {
+        if (trySearchDomain) {
+            final Promise<T> original = promise;
+            promise = new DefaultPromise<T>(parent.executor());
```

Comment:
@vietj use ´parent.executor.newPromise()´. No need to construct `DefaultPromise` explicitly